File: erlang/2.2/geom.erl
%%% -----------------------------------------------
%%% @author anders pearson
%%% @doc simple geometry module
%%% @end
%%% -----------------------------------------------
-module(geom).
-export([area/2]).

%% @spec area(X::integer(), Y::integer()) -> integer()
%% @doc calculates area. duh.
area(X, Y) ->
		X * Y.

File: elixir/2.3/geom.ex
defmodule Geom do
	@moduledoc """
	simple geometry module. done as an exercise for Elixir Etudes
	"""

	@doc """
	Calculates the area of a rectangle
	"""
	def area(width \\ 1, height \\ 1) do
		width * height
	end
end
